src/app/app.component.ts

```diff
@@ -1,13 +1,34 @@
-import {Component} from '@angular/core';
-import {TreeItem} from "./interfaces/TreeItem";
-import {TREE_ITEMS} from "./consts/TREE_ITEMS";
+import {Component, OnInit} from '@angular/core';
+import {TreeItemInterface} from "./models/tree-item.interface";
+import {TreeItems} from "./constants/tree-items";
+import {FormBuilder, FormGroup} from "@angular/forms";
+import {NodeBuilder} from "./services/node-builder.service";
+import {Node} from "./models/node";
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
 
-export class AppComponent {
-  items: TreeItem[] = TREE_ITEMS;
+export class AppComponent implements OnInit {
+  items: Node[] = this.cs.convertItemToNode(TreeItems);
+  myform!: FormGroup;
+
+  constructor(private fb: FormBuilder,
+              private cs: NodeBuilder) {
+  }
+
+  ngOnInit(): void {
+    this.myform = this.fb.group({
```

Comment:
as i remember, you're using angular 14. If it's correct, let's use typing forms with interfaces